decomp/semantics/uds.py

```diff
@@ -81,20 +89,32 @@ def __init__(self,
             errmsg = 'split must be "train", "dev", or "test"'
             raise ValueError(errmsg)
 
-        all_built = all(s in self._corpus_paths
+        all_built = all(s in self._sentences_paths
+                        for s in ['train', 'dev', 'test'])
+        all_built = all_built and all(s in self._documents_paths
                         for s in ['train', 'dev', 'test'])
 
         self._graphs = {}
+        self._documents = {}
         
-        if graphs is None and split in self._corpus_paths:
-            fpath = self._corpus_paths[split]
-            corp_split = self.__class__.from_json(fpath)
-            self._graphs.update(corp_split._graphs)
+        # Documents should not be provided without graphs
+        if docs and not graphs:
```

Comment:
Should this be checking `not graphs` or `graphs is None`?

Reply:
For consistency with the other conditionals in the constructor, it may make more sense to check `is None`, but I don't think it would ever make a practical difference here.

Reply:
Let's change this to checking for a `NoneType` then. The generality of this check seems like a future bug waiting to happen, since Python is very promiscuous in what it's willing to cast to a boolean.